### ISSUE TITLE
Nested `~{0}~{}` doesn't work

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#904]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | -
| File format of the file you reverse (mandatory)      | -
| Architecture/bits of the file (mandatory)            | -
| `rizin -v` full output, **not truncated** (mandatory)  | https://github.com/rizinorg/rizin/commit/bed31a6173a7a3632395498e0c7386ca72505ba0

### Expected behavior

Nested `~{}~{}` should still format JSON and provide syntax highlight similar to how it's done in the unnested case.

```
[0x00004020]> iij~{0} # extract 0th JSON array element
{"ordinal":1,"bind":"GLOBAL","type":"FUNC","name":"__ctype_toupper_loc","plt":0}
[0x00004020]> iij~{2} # extract 2nd JSON array element
{"ordinal":3,"bind":"GLOBAL","type":"FUNC","name":"free","plt":0}
[0x00004020]> iij~{0}~{}
{
  "ordinal":1,
  "bind":"GLOBAL",
  "type":"FUNC",
  "name":"__ctype_toupper_loc",
  "plt":0
 }
[0x00004020]> iij~{2}~{}
{
  "ordinal":3,
  "bind":"GLOBAL",
  "type":"FUNC",
  "name":"free",
   "plt":0
}
[0x00004020]> 
```

### Actual behavior

```
[0x00004020]> iij~{0}
{"ordinal":1,"bind":"GLOBAL","type":"FUNC","name":"__ctype_toupper_loc","plt":0}
[0x00004020]> iij~{2}
{"ordinal":3,"bind":"GLOBAL","type":"FUNC","name":"free","plt":0}
[0x00004020]> iij~{0}~{}
{"ordinal":1,"bind":"GLOBAL","type":"FUNC","name":"__ctype_toupper_loc","plt":0}
[0x00004020]> iij~{2}~{}
{"ordinal":3,"bind":"GLOBAL","type":"FUNC","name":"free","plt":0}
[0x00004020]> 
```

**Comment by ret2libc** [comment]
I agree. I think grep syntax should be completely moved and handled by the parser. Right now almost everything after `~` is passed to rz_cons_grep, which does not support many of the nested things (even multi-word grep are supported just through a hack that rewrites them into a single grep with `,`).

**Comment by XVilka** [comment]
@ret2libc in fact, now with `:` modifiers, it makes sense to maybe use something like `:json` or similar instead of `~{}` for pretty-printing. What do you think?

**Comment by ret2libc** [comment]
> @ret2libc in fact, now with `:` modifiers, it makes sense to maybe use something like `:json` or similar instead of `~{}` for pretty-printing. What do you think?

I think we want to use `:json` to get the json output mode in the future, but if you default to pretty-printing, how do you get the non-pretty output (e.g. for rz-pipe)?

**Comment by XVilka** [comment]
We could use `:json:pretty` or something like that.

**Comment by XVilka** [comment]
We should move `grep` feature out of the RzCons and to the RzCore.

**Comment by ret2libc** [comment]
> We should move `grep` feature out of the RzCons and to the RzCore.

I'm not sure.... I think there are a lot of things you need to do inside RzCons (though I haven't looked at the code in detail).
